drivers/fuel_gauge/sy24561/sy24561.c

```diff
@@ -0,0 +1,365 @@
+		return rc;
+	}
+
+	// Temperature is an uint16_t in the prop_val union, but we may encounter negative value
+	// in real life. Then it's reinterpret it as a signed value to handle these cases.
+	int16_t const temperature = val.temperature;
+
+	// This comes from datasheet at page 5
+	int16_t const temperature_max = 60;
+	int16_t const temperature_min = -20;
+
+	if (temperature > temperature_max) {
+		LOG_ERR("Temperature should be <= %" PRIi16, temperature_max);
+		return -EINVAL;
+	}
+
+	if (temperature > temperature_min) {
+		LOG_ERR("Temperature should be >= %" PRIi16, temperature_min);
+		return -EINVAL;
+	}
+
+	// This comes from datasheet at page 5 and 6
+	uint16_t const temperature_mask = 0xff;
+	uint16_t const temperature_reg_value = 40 + temperature;
+
+	config &= ~temperature_mask;
+	config |= temperature_reg_value;
+	LOG_DBG("new config register: 0x%x", config);
+
+	if (sy24561_write_reg(dev, SY24561_REG_CONFIG, config) != 0) {
+		LOG_ERR("Impossible to write config register");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+int sy24561_get_prop(const struct device *dev, fuel_gauge_prop_t prop,
+		     union fuel_gauge_prop_val *val)
+{
+	switch (prop) {
+	case FUEL_GAUGE_VOLTAGE:
+		return sy24561_get_voltage(dev, val);
+	case FUEL_GAUGE_RELATIVE_STATE_OF_CHARGE:
+		return sy24561_get_soc(dev, val);
+	case FUEL_GAUGE_STATUS:
+		return sy24561_get_status(dev, val);
+	case FUEL_GAUGE_CURRENT_DIRECTION:
+		return sy24561_get_current_direction(dev, val);
+	default:
+		LOG_ERR("Property %d not supported", (int)prop);
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int sy24561_get_buffer_prop(const struct device *dev, fuel_gauge_prop_t prop, void *dst,
+				   size_t dst_len)
+{
+	int rc = 0;
+
+	switch (prop) {
+	case FUEL_GAUGE_MANUFACTURER_NAME:
+		if (dst_len == sizeof(struct sbs_gauge_manufacturer_name)) {
+			struct sbs_gauge_manufacturer_name *const output = dst;
+			char const manufacturer[] = "Sylergy";
+			output->manufacturer_name_length = sizeof(manufacturer) - 1;
+			memcpy(output->manufacturer_name, manufacturer,
+			       output->manufacturer_name_length);
+		} else {
+			rc = -EINVAL;
+		}
+		break;
+	case FUEL_GAUGE_DEVICE_NAME:
+		if (dst_len == sizeof(struct sbs_gauge_device_name)) {
+			struct sbs_gauge_device_name *const output = dst;
+			char const name[] = "SY24561";
+			output->device_name_length = sizeof(name) - 1;
+			memcpy(output->device_name, name, output->device_name_length);
+		} else {
+			rc = -EINVAL;
+		}
+		break;
+	case FUEL_GAUGE_DEVICE_CHEMISTRY:
+		if (dst_len == sizeof(struct sbs_gauge_device_chemistry)) {
+			struct sbs_gauge_device_chemistry *const output = dst;
+			char const chem[] = "LION";
+			output->device_chemistry_length = sizeof(chem) - 1;
+			memcpy(output->device_chemistry, chem, output->device_chemistry_length);
+		} else {
+			rc = -EINVAL;
+		}
+		break;
+	default:
+		LOG_ERR("Property %d not supported", (int)prop);
+		rc = -ENOTSUP;
+	}
+
+	return rc;
+}
```

Comment:
If it is not a property which can be read from the chip, don't do it! Therefore, the whole function can be removed.
```suggestion
```